### PR TITLE
ROX-11862: Remove redundant conversion layer (ManagedCentralCR)

### PR DESCRIPTION
**Pull request by kovayur**
## Description
This is a refactoring PR which removes the unnecessary conversion layer - ManagedCervice CR. We don't have such CRD.

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Documentation added if necessary
- [ ] CI and all relevant tests are passing

## Test manual
Tested on a dedicated OSD cluster

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development
make verify lint binary test test/integration
```

### CODE REVIEW

**Comment by kovayur** [comment]
Current dependencies on/for this PR:
* main
  * **PR #237** <a href="https://app.graphite.dev/github/pr/stackrox/acs-fleet-manager/237" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/acs-fleet-manager/237?utm_source=stack-comment).

**Review by johannes94 — APPROVED** [review]
LGTM

**Comment by mtesseract** [comment]
Hi, even though I welcome such a simplification PR I do wonder if this is going in the right direction. I am saying this because I vaguely remember @SimonBaeumer bringing up the discussion point that our fleetshard-sync *is* already behaving mostly like an operator. My understanding is that it *could* have architectural benefits to introduce such CRs and keep up a distinction between the different presentations, even though I cannot give an concrete example at this point.

Am I missing some context here? I just want to make sure that we don't remove anything at this point which would turn out to be useful in a few weeks or months.

**Comment by kovayur** [comment]
Hi @mtesseract, thanks for sharing your concerns. TBH I don't see any reason why we should use the CR on the **fleet-manager** even though it may be used in the fleetshard sync in the future. I'd like follow the YAGNI approach and revert the deleted `types.go` when it become required. Is it okay for you?

File: internal/dinosaur/pkg/services/dinosaur.go [hunk 1]
```diff
@@ -823,65 +812,6 @@ func (k *dinosaurService) ListDinosaursWithRoutesNotCreated() ([]*dbapi.CentralR
...
-				TLS: &manageddinosaur.TLSSpec{
-					Cert: dinosaurConfig.DinosaurTLSCert,
-					Key:  dinosaurConfig.DinosaurTLSKey,
-				},
-			},
-			Versions: manageddinosaur.VersionsSpec{
-				Dinosaur:         dinosaurRequest.DesiredCentralVersion,
-				DinosaurOperator: dinosaurRequest.DesiredCentralOperatorVersion,
-			},
-			Deleted: dinosaurRequest.Status == constants2.DinosaurRequestStatusDeprovision.String(),
```
Comment (mtesseract) [inline internal/dinosaur/pkg/services/dinosaur.go:862]: This `Deleted` flag is not representable in the new model anymore, right?

Comment (kovayur) [reply internal/dinosaur/pkg/services/dinosaur.go:862]: Right, we rely on `deletedTimestamp` now

File: internal/dinosaur/pkg/presenters/managedcentral.go [hunk 2]
```diff
@@ -25,38 +26,51 @@ const (
...
+				from.Owner,
+			},
 			Auth: private.ManagedCentralAllOfSpecAuth{
-				ClientSecret: from.Spec.Auth.ClientSecret,
-				ClientId:     from.Spec.Auth.ClientID,
-				OwnerUserId:  from.Spec.Auth.OwnerUserID,
-				OwnerOrgId:   from.Spec.Auth.OwnerOrgID,
-				Issuer:       from.Spec.Auth.Issuer,
+				ClientSecret: c.centralConfig.RhSsoClientSecret, // pragma: allowlist secret
+				// TODO: ROX-11593: make part of centralConfig
```
[the remark this answers is not in the document]
Comment (kovayur) [reply internal/dinosaur/pkg/presenters/managedcentral.go]: This was copied from another place, I've changed the format

**Comment by mtesseract** [comment]
> 

Discussed offline, no objection.

**Comment by mtesseract** [comment]
> Hi @mtesseract, thanks for sharing your concerns. TBH I don't see any reason why we should use the CR on the **fleet-manager** even though it may be useful in the fleetshard sync in the future. I'd like follow the YAGNI approach and revert the deleted `types.go` when it become required. It shouldn't be difficult after all. Is it okay for you?

Discussed offline, no objection.

**Pull request merged**